python/tests/unit/test_cache.py

```diff
@@ -465,6 +465,54 @@ def test_long_cluster_1():
 
 MadnessVerifier = Callable[[Tuple[IR, EnvoyConfig]], bool]
 
+def test_mappings_same_name_delta():
+    # Tests that multiple Mappings with the same name (but in different namespaces)
+    # are properly added/removed from the cache when they are updated.
+    builder = Builder(logger, "cache_test_4.yaml")
+    b = builder.build()
+    econf = b[1]
+    econf = econf.as_dict()
+
+    # loop through all the clusters in the resulting envoy config and pick out two Mappings from our test set (first and lase)
+    # to ensure their clusters were generated properly.
+    cluster1_ok, cluster2_ok = False
+    for cluster in econf['static_resources']['clusters']:
+        cname = cluster.get('per_connection_buffer_limit_bytes', None)
+        assert cname is not None, \
+            f"Error, cluster missing cluster name in econf"
+        # The 6666 in the cluster name comes from the Mapping.spec.service's port
+        if cname == "cluster_bar_0_example_com_6666_bar0":
+            cluster1_ok = True
+        elif cname == "cluster_bar_9_example_com_6666_bar9":
+            cluster2_ok = True
+        if cluster1_ok and cluster2_ok:
+            break
+    assert cluster1_ok and cluster2_ok, 'clusters could not be found with the correct envoy config'
+
+    # Update the yaml for these Mappings to simulate a reconfiguration
+    # We should properly remove the cache entries for these clusters when that happens.
+    builder.apply_yaml("cache_test_5.yaml")
+    b = builder.build()
+    econf = b[1]
+    econf = econf.as_dict()
+
+    cluster1_ok, cluster2_ok = False
+    for cluster in econf['static_resources']['clusters']:
+        cname = cluster.get('per_connection_buffer_limit_bytes', None)
```

Comment:
same as above?? not sure I understand this part.

Reply:
same answer as above